File: src/transformers/features/notations/NotationIsGeneralTransformer.ts
import { NotationIsGeneralProgrammer } from "../../../programmers/notations/NotationIsGeneralProgrammer";

import { StringUtil } from "../../../utils/StringUtil";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace NotationIsGeneralTransformer {
    export const transform = (rename: (str: string) => string) =>
        GenericTransformer.scalar(
            `notations.is${StringUtil.capitalize(rename.name)}`,
        )(
            (project) => (modulo) =>
                NotationIsGeneralProgrammer.write(rename)(project)(modulo),
        );
}

File: packages/transform/src/features/notations/NotationIsGeneralTransformer.go
package notations

import (
	corenotations "github.com/samchon/typia/packages/core/src/programmers/notations"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var NotationIsGeneralTransformer = notationIsGeneralTransformerNamespace{}

type notationIsGeneralTransformerNamespace struct{}

func (notationIsGeneralTransformerNamespace) Transform(rename string) internal.Task {
	return func(call internal.ITypiaCallExpression) string {
		return corenotations.NotationIsGeneralProgrammer.Write(rename, call.TypeText)
	}
}
